src/librustc_mir/hair/pattern/check_match.rs

```diff
@@ -121,7 +123,9 @@ impl PatCtxt<'_, '_> {
 impl<'tcx> MatchVisitor<'_, 'tcx> {
     fn check_patterns(&mut self, has_guard: bool, pat: &Pat) {
         check_legality_of_move_bindings(self, has_guard, pat);
-        check_legality_of_bindings_in_at_patterns(self, pat);
+        if !self.tcx.features().bindings_after_at {
+            check_legality_of_bindings_in_at_patterns(self, pat);
```

Comment:
Is this just an efficiency hack, to avoid the work of doing the lint when the feature is enabled?

Reply:
Not sure what lint you are referring to. I've feature gated things this way so that when we want to stabilize, it will be clear that this function `check_legality_of_bindings_in_at_patterns` should just be whole-sale removed. It's probably also more efficient but that's secondary. ;)